enhancements/monitoring/monitoring-stack-operator.md

```diff
@@ -0,0 +1,490 @@
+
+### Non-Goals
+
+* Access controls for managed service owners for metrics from different managed services. For example, there are currently no requirements for owners of managed services A to be prohibited from seeing metrics from managed service B, and vice versa.
+* Expose managed service metrics to end-users. We know e.g. OCS addon wants this, but this can be achieved on top of the proposed solution using client or backend side filtering for data isolation if needed (e.g. prom-label-proxy). The monitoring team can do this in the next iteration.
+* Integrate managed service metrics into the OpenShift console. We know OCS wants to integrate with OCP Console, but we assume the console team can help support switching between Prometheus-compatible data sources.
+* Solve Platform Monitoring SPOF, scalability and overhead. This will be targeted in a follow-up enhancement.
+* Allow logging and tracing capabilities. Other solutions should tackle this (designed by Logging and Tracing teams.)
+
+## Proposal
+
+### Personas
+
+* OSD/Platform SRE
+* MTSRE
+* RHOBS Devs
+* MTO
+* Customer: User of Managed Service and OSD
+
+### User Stories
+
+#### Story 1
+
+As a MTSRE/MTO, I would like to define ServiceMonitors that will scrape metrics from my managed service by a Prometheus-compatible system.
+
+#### Story 2
+
+As an MTSRE/MTO, I would like to query SaaS/Addon metrics using PromQL to understand the health or troubleshoot the performance of my managed service.
+
+#### Story 3
+
+As an MTSRE/MTO, I would like to create alerting and recording rules for Prometheus metrics exposed by my managed service.
+
+#### Story 4
+
+As an MTSRE, I would like to configure routing for alerts defined on Prometheus metrics exposed by my SaaS/Addon.
+
+#### Story 5
+
+As an MTO, I would like to query certain metrics for my service across all clusters in which the managed service is deployed or used.
+
+#### Story 6
+
+As an MTO, I would like to create dashboards for certain metrics across all clusters where the SaaS/Addon is deployed or used.
+
+#### Story 7
+
+As a Customer, I can use UWM without seeing Addons metrics, alerting or recording configurations.
+
+### Implementation Details/Notes/Constraints
+
+Since UWM and PM are part of Openshift Core, we would like to avoid increasing the complexity and introducing potential bugs for customers by extending those monitoring stacks. Because of this, we propose developing a new stack that will be able to support the managed service use-case while staying compatible with upcoming use-cases such as Hypershift or [KCP](https://github.com/kcp-dev/kcp).
+
+#### Overview
+
+We propose developing a new open-source operator with a working name, “MonitoringStack Operator” (official name proposals are welcome!) that manages a `MonitoringStack` CRD and can be used to deploy an independent, prometheus-based stack for each managed service.
```

Comment:
Can we possibly please clarify here for the case of more than 1 addon in a cluster? What does `stack` here means, is this referring to the set of ServiceMonitors per addon only or the prom/alertm instances?